### ISSUE TITLE
bug when combining `cmt_star_cont` cpp comment `//` and `||`

### ISSUE THREAD

**Issue by CDanU** [uncrustify/uncrustify#1140]
I:
```cpp
/*
if (a
// c
|| b)
{
}
*/
```
O:
```cpp
/*
 * if (a
 * // c
 || b)
 ||{
 ||}
 */
```
E:
```cpp
/*
 * if (a
 * // c
 *     || b)
 * {
 * }
 */
```

<details><summary>DEBUG</summary>
<pre>
# Uncrustify-0.64-595-79ba971f
cmt_star_cont                   = true
# option(s) with 'not default' value: 1
#
# -=====-
# Line              Tag           Parent          Columns Br/Lvl/pp     Flag   Nl  Text
#   1>    COMMENT_MULTI[   COMMENT_WHOLE][  1/  1/  3/  0][0/0/0][         0][6-0] /*␤if (a␤// c␤|| b)␤{␤}␤*/
#   7>          NEWLINE[            NONE][  3/  3/  1/  0][0/0/0][         0][1-0]
# -=====-
<pre>
</details>

**Comment by CDanU** [comment]
This behavior is caused by the if block in [output.cpp:1412](https://github.com/uncrustify/uncrustify/blob/79ba971f3a78f00a2f88ee3d2bd15682f55d806c/src/output.cpp#L1412). 
If `line[0] != '|'` is removed everything works as expected. 
The problem is that I don't know why all of those characters (except the `*`) are being checked for.

**Comment by guy-maurel** [comment]
| is a lead character, as defined at` output.cpp@cmt_parse_lead`:
```
      else if (strchr("*|\\#+", line[len]) == nullptr)
      {
         break;
```
At line 4 of the example there is a lead character |

This is correct, this is not a bug.

**Comment by guy-maurel** [comment]
Using uncrustify command option` -L 83` one can see:
```
output_comment_multi:1209 set cont_text to '* '
/*
 * if (a
 * // c
output_comment_multi:1453 set cont_text to '||'
 || b)
 ||{
 ||}
output_comment_multi:1453 set cont_text to '*'
 */
```

**Comment by CDanU** [comment]
Yes I understand that, what I don't get is where such a lead character occurs. 
It does not seem to be C++ related but is causes the above unwanted behavior.

**Comment by guy-maurel** [comment]
Is it not related to doxygen?
